### PR TITLE
8285405: add test and check for negative argument to HashMap::newHashMap et al

### PR DESCRIPTION
**Pull request by jaikiran**
(This is a recreation of a previous pull request which had received some reviews https://github.com/openjdk/jdk/pull/9036. I had to delete that personal branch and recreate it due to some git issues)

Can I please get a review of this change which addresses https://bugs.openjdk.java.net/browse/JDK-8285405?

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8285405](https://bugs.openjdk.org/browse/JDK-8285405): add test and check for negative argument to HashMap::newHashMap et al

### Reviewers
 * [Chris Hegarty](https://openjdk.org/census#chegar) (@ChrisHegarty - **Reviewer**) ⚠️ Review applies to [ee516ffd](https://git.openjdk.org/jdk/pull/9806/files/ee516ffd65729785d6aba7ca3fc03ba11069ef39)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [ee516ffd](https://git.openjdk.org/jdk/pull/9806/files/ee516ffd65729785d6aba7ca3fc03ba11069ef39)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Stuart Marks](https://openjdk.org/census#smarks) (@stuart-marks - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9806/head:pull/9806` \
`$ git checkout pull/9806`

Update a local copy of the PR: \
`$ git checkout pull/9806` \
`$ git pull https://git.openjdk.org/jdk pull/9806/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9806`

View PR using the GUI difftool: \
`$ git pr show -t 9806`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9806.diff">https://git.openjdk.org/jdk/pull/9806.diff</a>

</details>

### CODE REVIEW

**Comment by jaikiran** [comment]
Hello Stuart (@stuart-marks), as suggested by you in the pull request that you previously reviewed, I've now moved the tests to `HashMap/WhiteBoxResizeTest` and it now tests the new APIs using (only) `-1` as the negative input.

**Review by ChrisHegarty — APPROVED** [review]
LGTM

File: test/jdk/java/util/HashMap/WhiteBoxResizeTest.java [hunk 1]
```diff
@@ -426,4 +426,34 @@ public int capacity() {
...
+    public void testNegativeNumMappings(final IntFunction<?> method, final String methodName) {
+        try {
+            method.apply(-1);
+            // expected to fail but didn't
+            throw new AssertionError(methodName + " was expected to throw" +
+                    " IllegalArgumentException for param -1, but didn't");
+        } catch (IllegalArgumentException expected) {
+            // received the expected exception
+        }
+    }
```
Comment (naotoj) [inline test/jdk/java/util/HashMap/WhiteBoxResizeTest.java:458]: Could utilize `assertThrows`, instead of try-catch.

**Review by naotoj — APPROVED** [review]
Looks good as it stands.

**Review by LanceAndersen — CHANGES_REQUESTED** [review]

**Comment by jaikiran** [comment]
Thank you Chris, Lance and Naoto for the reviews.

I've now updated the PR to use `assertThrows` as suggested by Naoto and Lance.

**Review by LanceAndersen — APPROVED** [review]

**Review by stuart-marks — APPROVED** [review]
Thanks for following up on this. Looks good.

**Comment by jaikiran** [comment]
Thank you everyone for the reviews.

**Pull request closed without merging**